go-controller/pkg/ovn/policy.go

```diff
@@ -747,21 +746,26 @@ func (oc *Controller) deleteNetworkPolicy(policy *knet.NetworkPolicy) {
 
 	np := oc.deleteNetworkPolicyLocked(policy)
 	if np == nil {
+		klog.V(5).Infof("Failed to get namespace lock when deleting policy %s in namespace %s",
+			policy.Name, policy.Namespace)
 		return
 	}
 	defer np.Unlock()
 
-	// We should now stop all the handlers go routines.
+	oc.cleanNPentities(np)
+}
+
+func (oc *Controller) cleanNPentities(np *namespacePolicy) {
```

Comment:
@astoycos this is really a function to clean up a single namespacePolicy, so how about `destroyNamespacePolicy()`?

Reply:
Yeah I couldn't come up with a good name... I like this suggestion